diff -ruN mingw/etc/profile dreamsdk/etc/profile
--- mingw/etc/profile	2009-07-11 20:34:34 +0100
+++ dreamsdk/etc/profile	2022-08-12 13:48:00 +0100
@@ -4,7 +4,7 @@
 # 
 #         File:	profile
 #  Description:	Shell environment initialization script
-# Last Revised:	2002.05.04
+# Last Revised:	2021.11.28 (DreamSDK)
 
 if [ -z "$MSYSTEM" ]; then
   MSYSTEM=MINGW32
@@ -56,5 +56,85 @@
 
 alias clear=clsb
 
-cd "$HOME"
+################################################################################
+#    ___                     _______  __ __
+#   / _ \_______ ___ ___ _  / __/ _ \/ //_/
+#  / // / __/ -_) _ \`/  ' \_\ \/ // /  ,<
+# /____/_/  \__/\_,_/_/_/_/___/____/_/|_|
+#                                        
+################################################################################
 
+# Handle external updated Wget
+export WGETRC=/etc/wgetrc
+
+# Force GNU compilers messages in English language
+# This avoid some encoding problems in the console
+export LANG=en
+
+################################################################################
+
+# DreamSDK base directory
+export dreamsdk_base=/opt/dreamsdk
+export PATH="${PATH}:${dreamsdk_base}"
+
+# Toolchains base directory
+export toolchains_base=/opt/toolchains/dc
+
+# DreamSDK configuration file
+export dreamsdk_conf=/etc/dreamsdk/dreamsdk.conf
+
+# Dreamcast Tool configuration file
+export dctool_conf=/etc/dreamsdk/dc-tool.conf
+
+################################################################################
+
+# KallistiOS initialization script
+export environ=$toolchains_base/kos/environ.sh
+
+# KallistiOS utilities directory
+export kos_utilities=$toolchains_base/kos/utils
+
+# Add additional directories in the PATH
+export PATH="${PATH}:${dreamsdk_base}/addons"
+export PATH="${PATH}:${dreamsdk_base}/helpers"
+export PATH="${PATH}:${dreamsdk_base}/scripts"
+
+# Initialize KallistiOS
+export kos_initialized=0
+if [ -f "${environ}" ]; then
+  source "${environ}"
+  export kos_initialized=1	  
+fi
+
+# Add all additional KallistiOS utilities in PATH
+if [ $kos_initialized -eq 1 ]; then
+  for d in ${kos_utilities}/*/ ; do
+    [ -L "${d%/}" ] && continue
+    export PATH="${PATH}:$d"
+  done
+fi
+
+# Change to the passed directory if needed...
+if [ -n "$_WORKING_DIRECTORY" ]; then
+  cd "$_WORKING_DIRECTORY"  
+fi
+
+# By default, MSYS is always switching to the home directory.
+# We don't want this, as we will allow to open a shell in a specific folder and
+# run some commands from there directly from the prompt.
+if [ -z "$_EXTERNAL_COMMAND" ]; then
+  # Default usage (as normal MSYS instance)
+  if [ -z "$_WORKING_DIRECTORY" ]; then    
+    cd "$HOME"
+  fi
+  
+  # Display DreamSDK welcome banner
+  dreamsdk-banner  
+else
+  # Command passed to the launcher
+  $_EXTERNAL_COMMAND
+  if [ -n "$_EXITCODE" ]; then
+    echo $? > "$_EXITCODE"
+  fi
+  logout
+fi
