--- /etc/profile	2009-07-11 18:34:34 +0000
+++ /etc/profile.dcsdk	2018-10-29 17:51:45 +0000
@@ -56,5 +56,36 @@
 
 alias clear=clsb
 
-cd "$HOME"
+#cd "$HOME"
 
+# --- DreamSDK Additions ---
+
+# Toolchain base directory
+export toolchains_base=/opt/toolchains/dc
+
+# Force GNU compilers to be in English language
+export LANG=en
+
+# Initialize KallistiOS
+environ=$DCSDK_BASE/kos/environ.sh
+if [ -f ${environ} ]; then
+  source ${environ}
+fi
+
+# Add additional items in the PATH
+export PATH="${PATH}:/opt/dcsdk/helpers/"
+
+# By default, MSYS is always switching to the home directory.
+# We don't want this, as we will allow to open a shell in a specific folder and
+# run some commands from there directly from the prompt.
+if [ -z "$_EXTERNAL_COMMAND" ]; then
+  # Default usage (as normal MSYS instance)
+  cd "$HOME"
+else
+  # Command passed to the dcsdk-launcher
+  $_EXTERNAL_COMMAND
+  if [ -z "$_AUTOMATED_CALL" ]; then
+    pause
+  fi
+  logout
+fi
