--- profile	2009-07-11 18:34:34 +0000
+++ profile.dcsdk	2018-11-07 14:01:08 +0000
@@ -56,5 +56,50 @@
 
 alias clear=clsb
 
-cd "$HOME"
+#cd "$HOME"
 
+### DreamSDK ###################################################################
+
+# Toolchains base directory
+export toolchains_base=/opt/toolchains/dc
+
+# KallistiOS initialization script
+export environ=$toolchains_base/kos/environ.sh
+
+# DreamSDK base directory
+export dreamsdk_base=/opt/dreamsdk
+
+# DreamSDK configuration file
+export dreamsdk_conf=/etc/dreamsdk/dreamsdk.conf
+
+# Add additional items in the PATH
+export PATH="${PATH}:${dreamsdk_base}/scripts/"
+
+# Force GNU compilers messages in English language
+# This avoid some encoding problems in the console.
+export LANG=en
+
+# Initialize KallistiOS
+export kos_initialized=0
+if [ -f ${environ} ]; then
+  source ${environ}
+  export kos_initialized=1	  
+fi
+
+# By default, MSYS is always switching to the home directory.
+# We don't want this, as we will allow to open a shell in a specific folder and
+# run some commands from there directly from the prompt.
+if [ -z "$_EXTERNAL_COMMAND" ]; then
+  # Default usage (as normal MSYS instance)
+  cd "$HOME"
+  
+  # Display version and banner
+  dreamsdk-version  
+else
+  # Command passed to the launcher
+  $_EXTERNAL_COMMAND
+  if [ "$_AUTOMATED_CALL" -eq 0 ]; then
+    pause
+  fi
+  logout
+fi
